pallets/moonbeam-lazy-migrations/src/lib.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2024 Moonbeam foundation
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # Lazy Migration Pallet
+
+#![allow(non_camel_case_types)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+use frame_support::pallet;
+
+pub use pallet::*;
+
+#[pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+	use sp_core::ConstBool;
+
+	const INTERMEDIATES_NODES_SIZE: u64 = 4096;
+	const MAX_LOCAL_ASSETS_STORAGE_ENTRY_SIZE: u64 =
+		(/* biggest key on moonbeam */136) + (/* biggest value on moonbeam */142);
+
+	/// Pallet for multi block migrations
+	#[pallet::pallet]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	#[pallet::storage]
+	/// If true, it means that LocalAssets storage has been removed.
+	pub(crate) type LocalAssetsMigrationCompleted<T: Config> =
+		StorageValue<_, bool, ValueQuery, ConstBool<false>>;
```

Comment:
```suggestion
		StorageValue<_, bool, ValueQuery>;
```

The default default value is already false